common/json.c

```diff
@@ -292,24 +292,36 @@ jsmntok_t *json_tok_copy(const tal_t *ctx, const jsmntok_t *tok)
 	return tal_dup_arr(ctx, jsmntok_t, tok, json_next(tok) - tok, 0);
 }
 
-void json_tok_remove(jsmntok_t **tokens, jsmntok_t *tok, size_t num)
+void json_tok_remove(jsmntok_t **tokens,
+		     jsmntok_t *obj_or_array, const jsmntok_t *tok, size_t num)
 {
-	assert(*tokens);
-	assert((*tokens)->type == JSMN_ARRAY || (*tokens)->type == JSMN_OBJECT);
 	const jsmntok_t *src = tok;
 	const jsmntok_t *end = json_next(*tokens);
-	jsmntok_t *dest = tok;
+	jsmntok_t *dest = *tokens + (tok - *tokens);
```

Comment:
Ahh, this removes the const qualifier!  I had to compile and double check for myself.